packages/react-charting/src/components/DonutChart/DonutChart.base.tsx

```diff
@@ -289,10 +289,10 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
 
   private _valueInsideDonut(valueInsideDonut: string | number | undefined, data: IChartDataPoint[]) {
     const highlightedLegend = this._getHighlightedLegend();
-    if (valueInsideDonut !== undefined && highlightedLegend !== '' && !this.state.showHover) {
+    if (valueInsideDonut !== undefined && (highlightedLegend !== '' || this.state.showHover)) {
       let legendValue = valueInsideDonut;
       data!.map((point: IChartDataPoint, index: number) => {
-        if (point.legend === highlightedLegend) {
+        if (point.legend === highlightedLegend || (this.state.showHover && point.legend === this.state.legend)) {
```

Comment:
does this value represent the hovered element?

Reply:
yes, _hoverCallback handler is called on hovering over a particular pie of the donut, which sets the state.legend value. And _getHighlightedLegend() already sets the highlightedLegend based on whether a legend is selected or hovered (active legend).